evm/vm/base.py

```diff
@@ -122,6 +133,49 @@ def get_block_reward(self, block_number):
     def get_nephew_reward(self, block_number):
         return NEPHEW_REWARD
 
+    def import_block(self, block):
+        parent_header = self.evm.get_block_header_by_hash(block.header.parent_hash)
+        init_header = self.create_header_from_parent(parent_header)
+        evm = type(self.evm)(self.db, init_header)
+        vm = type(self)(evm, self.db)
```

Comment:
Yeah, that will have to be sorted out, ideally sooner rather than later, but right now I'm not sure how we should go about it. I hope the right abstractions will become more clear as I get more familiar with this part of the code.

Also, another abstraction that is not ideal is for block validation, which is currently done exclusively at the block level (in `Block.validate()`), but AIUI part of the block validation involves looking up a block's ancestor chain (up to the max uncle reference depth) as well as the uncles included in recent blocks.